### PR TITLE
DeploymentConfig to Deployment

### LINKED ISSUE #3873: DeploymentConfig's are deprecated
- We should be using Deployments going forward
- No ETA on when DC's will be removed, but it's recommended we start our transition now (from Platform Community 2024-08-27)
- https://access.redhat.com/articles/7041372 
- https://developers.redhat.com/learning/learn:openshift:replace-deprecated-deploymentconfigs-deployments/resource/resources:convert-deploymentconfig-deployment?mc_cid=1a844df827

Image

### PR DESCRIPTION
**Pull request by brettedw**
- Convert all DeploymentConfigs to Deployments
- Changes deploy bash script
   - Several of the commands previously used don't work with deployments
   - bash script is far more simple now and I believe functions similarly. Major change being rollouts now happen due to image changes or config changes
- Adds triggers to deployments. Now whenever the api or web image changes, it will trigger a new rollout
- Redis DC also converted to Deployment, still needs to be manually deployed
- Also changes ZAP check to the correct URL
- Closes #3873 

# Test Links:
[Landing Page](https://wps-pr-3935-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3935-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3935-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-3935-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3935-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3935-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3935-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3935-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: openshift/templates/redis.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    metadata:
+      labels:
+        name: wps-redis
+    spec:
+      automountServiceAccountToken: false
+      volumes:
+        - name: wps-redis-data
+          emptyDir: {}
+      containers:
+        - name: redis
```
[the remark this answers is not in the document]
Comment (brettedw) [reply openshift/templates/redis.yaml:44]: I think I took care of this but the warning is being persistent

**Review by conbrad — APPROVED** [review]
Nice work!

File: openshift/templates/redis.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgboss) [reply openshift/templates/redis.yaml:44]: Perhaps it wants CPU request/limit as well?

**Review by dgboss — APPROVED** [review]
Nice work!

**Pull request merged**